File: cpp/codeforces/burning_midnight_oil/burning_midnight_oil.cpp
#include <iostream>
#include <cmath>
using namespace std;

long long n, k;

bool less_than_n(long long v, long long k) {
    long long exp = 0;
    long long sum = 0;
    while(true) {
        long long tmp = floor(v / (pow(k, exp)));
        if(tmp <= 0) break;
        sum += tmp;
        exp++;
    }
    return sum < n;
}

int main() {
    cin >> n >> k;

    long long lo(0), hi(n);
    while(lo + 1 < hi) {
        long long mid = (long long)((lo + hi) / 2);
        if(less_than_n(mid, k)) lo = mid;
        else hi = mid;
    }
    cout << hi;
}

File: python/codeforces/burning_midnight_oil/burning_midnight_oil.py
from math import floor
n, k = (map(int, input().split()))

def less_than_n(v, k):
    exp = 0
    total = 0
    while True:
        tmp = floor(v / (k**exp))
        if tmp <= 0: break
        total += tmp
        exp += 1
    return total < n
        
lo = 0
hi = n
while lo + 1 < hi:
    mid = int((lo + hi)/2)
    if less_than_n(mid, k): lo = mid
    else: hi = mid

print(hi)

